examples/scale/scenarios/backfill/backfill.go

```diff
@@ -0,0 +1,414 @@
+
+		if !ok || len(tickets) == 0 {
+			// no tickets in pool
+			continue
+		}
+
+		for _, b := range backfills {
+			l := len(tickets)
+
+			if l == 0 {
+				// no tickets left
+				break
+			}
+
+			if l > s.TicketsPerMatch && s.TicketsPerMatch > 0 {
+				l = s.TicketsPerMatch
+			}
+
+			matches = append(matches, &pb.Match{
+				MatchId:       fmt.Sprintf("profile-%v-time-%v-%v", p.GetName(), time.Now().Format("2006-01-02T15:04:05.00"), len(matches)),
+				Tickets:       tickets[0:l],
+				MatchProfile:  p.GetName(),
+				MatchFunction: "backfill",
+				Backfill:      b,
+			})
+			tickets = tickets[l:]
+		}
+	}
+
+	return matches, nil
+}
+
+func (s *BackfillScenario) Evaluate(stream pb.Evaluator_EvaluateServer) error {
+	tickets := map[string]struct{}{}
+	backfills := map[string]struct{}{}
+	matchIds := []string{}
+
+outer:
+	for {
+		req, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("failed to read evaluator input stream: %w", err)
+		}
+
+		m := req.GetMatch()
+
+		if _, ok := backfills[m.Backfill.Id]; ok {
+			continue outer
+		}
+
+		for _, t := range m.Tickets {
+			if _, ok := tickets[t.Id]; ok {
+				continue outer
+			}
+		}
+
+		for _, t := range m.Tickets {
+			tickets[t.Id] = struct{}{}
+		}
+
+		matchIds = append(matchIds, m.GetMatchId())
+	}
+
+	for _, id := range matchIds {
+		err := stream.Send(&pb.EvaluateResponse{MatchId: id})
+		if err != nil {
+			return fmt.Errorf("failed to sending evaluator output stream: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (s *BackfillScenario) Backend() func(pb.BackendServiceClient, pb.FrontendServiceClient, *logrus.Entry) error {
+	return func(be pb.BackendServiceClient, fe pb.FrontendServiceClient, logger *logrus.Entry) error {
+		w := logger.Writer()
+		defer w.Close()
+
+		matchesToAcknowledge := make(chan *pb.Match, 60000)
+
+		for i := 0; i < s.GoroutinesAckTickets; i++ {
+			go runAcknowledgeBackfills(fe, matchesToAcknowledge, logger)
+		}
+
+		// Don't go faster than this, as it likely means that FetchMatches is throwing
+		// errors, and will continue doing so if queried very quickly.
+		for range time.Tick(time.Millisecond * 250) {
+			// Keep pulling matches from Open Match backend
+			profiles := s.Profiles()
+			var wg sync.WaitGroup
+
+			for _, p := range profiles {
+				wg.Add(1)
+				go func(wg *sync.WaitGroup, p *pb.MatchProfile) {
+					defer wg.Done()
+					runFetchMatches(be, p, matchesToAcknowledge, logger)
+				}(&wg, p)
+			}
+
+			// Wait for all profiles to complete before proceeding.
+			wg.Wait()
+			telemetry.RecordUnitMeasurement(context.Background(), metrics.Iterations)
+		}
+
+		return nil
+	}
+}
+
+func runAcknowledgeBackfills(fe pb.FrontendServiceClient, matchesToAcknowledge <-chan *pb.Match, logger *logrus.Entry) {
+	for m := range matchesToAcknowledge {
+		ids := []string{}
+		for _, t := range m.Tickets {
+			ids = append(ids, t.GetId())
+		}
+
+		err := doAcknowledgeBackfill(fe, m.Backfill.Id)
```

Comment:
We have only one `AcknowledgeBackfill` per Backfill returned by FetchMatches. In the real case there would be series of `AcknowledgeBackfill`. Say every minute.

Reply:
This is the fastest way to feel the match. So probably OK for now.